PTerm-Core/LibPTerm.class.st

```diff
@@ -56,34 +119,10 @@ LibPTerm class >> TIOCSWINSZ [
 	^21524
 ]
 
-{ #category : #compilation }
-LibPTerm class >> compile [
-	|cmd arch st fp libfile ofile|
-	fp := self mksource.
-	ofile := fp parent / 'libpterm.o'.
-	libfile := fp parent / 'libpterm.flib'.
-	arch := Smalltalk vm wordSize = 8 ifTrue: [ '-m64' ] ifFalse: [ '-m32' ].
-	cmd := 'gcc -c -fpic ', arch, ' "', fp pathString, '" -o "', ofile pathString, '"'.
-	Transcript show: 'Building: ', cmd; cr.
-	st := LibC system: cmd.
-	st = 0 ifFalse: [ ^self error:'LibC Cannot create object file' ].
-	cmd := 'gcc -shared ', arch, ' -o "',libfile pathString,'" "', ofile pathString, '"'.
-	Transcript show: 'Building: ', cmd; cr.
-	st := LibC system: cmd. 
-	st = 0 ifFalse: [ ^self error:'LibC Cannot create shared file' ].
-	fp exists ifTrue:[ fp delete ].
-	ofile asFileReference delete
-]
-
-{ #category : #compilation }
-LibPTerm class >> mksource [
-	|file stream|
-	file := (FileLocator localDirectory absolutePath / 'libpterm.c') asFileReference.
-	file exists ifTrue:[ file delete ].
-	stream := file writeStream.
-	stream nextPutAll: self source.
-	stream close.
-	^file
+{ #category : #initialization }
+LibPTerm class >> initialize [
+	IsLinux := Smalltalk os version beginsWith: 'linux'
```

Comment:
This will probably break MacOS support.  I suggest to rename the variable to `isPosix` and add MacOS support in OS version check.

Still need to be tested on all Posix-related changes on macOS

Reply:
> This will probably break MacOS support. I suggest to rename the variable to `isPosix`

Strictly speaking, that's not correct (either). IIRC, POSIX specifies only existence of various constants (see uses of `IsLinux`) but not their values. So on other systems the actual numerical values may differ. 

As for macOS, sure, should be done but I have no access to macOS...

Reply:
Yeah, my bad for fast thinking. In this case, we could do something link:

```st
POSIX_SPAWN_RESETIDS
  Smalltalk os isLinux ifTrue:[ ^ 16r01 ].
  Smalltalk os isMacOS ifTrue: [ ^....].
  self error: '(Yet) unsupported operating system'
```

Reply:
This was exactly my idea.